File: querygen/src/main/java/com/jtelaa/bwbot/querygen/util/GenericThread.java
package com.jtelaa.bwbot.querygen.util;

public class GenericThread extends Thread {

    // ------------------------- Thread Control

    /** Boolean to control the thread */
    public boolean run = true;

    /** Stops the thread */
    public synchronized void stopThread() { run = false; }

    /** Checks if the thread is ready */
    public synchronized boolean threadReady() { return run; }
    
}

File: querygen/src/main/java/com/jtelaa/bwbot/querygen/util/GenericThread.kt
package com.jtelaa.bwbot.querygen.util

import kotlin.jvm.Synchronized

/**
 * Abstractionn for the threads/processes in this application.
 *
 * @author Joseph
 */

open class GenericThread : Thread() {
    // ------------------------- Thread Control

    /** Boolean to control the thread  */
    var run = true

    /** Stops the thread  */
    @Synchronized
    fun stopThread() {
        run = false

    }

    /** Checks if the thread is ready  */
    @Synchronized
    fun threadReady(): Boolean {
        return run
        
    }
}